app/models/supplejack_api/concerns/privacyable.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module SupplejackApi
+  module Concerns
+    module Privacyable
+      extend ActiveSupport::Concern
+
+      included do
+        field :privacy, type: String,   default: 'public'
+
+        validates :privacy, inclusion: { in: %w[public hidden private] }
+        before_validation :set_default_privacy
+
+        # can't add this scope without breaking the model
+        # it must be an existing method in Mongoid
+        # scope :public,              -> { publicly_viewable }
```

Comment:
quite annoyed by that one :/

Reply:
`public` is also a keyword (it's an access modifier like `private` and `protected`, although it can be used as variables I think, so idk when it conflicts). You could also call it `unrestricted`? 🤷‍♂️  What you have is fine though 😄 

I was gonna suggest simplifying this into an `enum`, would then give you the validations, scopes and inquiry methods for free, although I don't think mongoid has an `enum` class method.

Reply:
Doesn't exist :/